src/vm/peimage.cpp

```diff
@@ -909,73 +909,141 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
     {
         _ASSERTE(HasID());
 
-        if (imageLayoutMask&PEImageLayout::LAYOUT_MAPPED)
-        {
-            PEImageLayout * pLoadLayout = NULL;
+        BOOL bIsMappedLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_MAPPED) != 0);
+        BOOL bIsFlatLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_FLAT) != 0);
 
-            if (m_bIsTrustedNativeImage || IsFile())
-            {
-                // For CoreCLR, try to load all files via LoadLibrary first. If LoadLibrary did not work, retry using 
-                // regular mapping - but not for native images.
-                pLoadLayout = PEImageLayout::Load(this, TRUE /* bNTSafeLoad */, m_bIsTrustedNativeImage /* bThrowOnError */);
```

Comment:
Done. The change is now represented with two commits.
The first splits `PEImage::GetLayoutInternal` into `PEImage::CreateLayoutMapped` and `PEImage::CreateLayoutFlat`, and the second changes way of mapping assemblies.

View of first commit without whitespace changes:
https://github.com/dotnet/coreclr/pull/10772/commits/70d27df6bd8724f25ae1df844c0c57505bc7b414?w=1